qiskit/ignis/verification/entanglement/parallelize.py

```diff
@@ -0,0 +1,449 @@
+
+
+'''
+In this package, there are three modules, linear.py,
+parallellize.py, and analysis.py. It may be more suitable
+to put these in Terra rather than Ignis. The most
+important module, parallellize.py provides methods to
+parallellize CNOT gates in the preparation of the GHZ State,
+which results in the GHZ State having a much higher fidelity
+then a normal "linear" CNOT gate preparation of the
+GHZ State. Additionally, there are methods within parallelize.py
+that configure different characterization tests for the
+GHZ State, including Multiple Quantum Coherence (MQC),
+Parity Oscillations (PO), and Quantum Tomography.
+'''
+
+from qiskit import *
+from qiskit.circuit import Parameter
+
+
+class BConfig:
+    '''
+    This class is used to create a GHZ circuit
+    with parallellized CNOT gates to increase fidelity
+    '''
+
+    def __init__(self, backend, indicator=True):
+        self.nodes = {}
+        self.backend = backend
+        self.cmap = backend.configuration().coupling_map
+        self.add_nodes()
+        self.indicator = indicator
+
+    def add_nodes(self):
+        '''
+        Adds nodes to the dictionary based on coupling map
+        '''
+        self.nodes = {}
+        for i in range(len(self.cmap_calib())):
+            self.nodes[self.cmap_calib()[i][0]] = []
+        for i in range(len(self.cmap_calib())):
+            self.nodes[self.cmap_calib()[i][0]].append(self.cmap_calib()[i][1])
+
+    def cmap_calib(self):
+        '''
+        Only intended for public devices (doubling and reversing
+        each item in coupling map), but useful to run anyway
+        '''
+        cmap_new = list(self.cmap)
+        for a in self.cmap:
+            if [a[1], a[0]] not in self.cmap:
+                cmap_new.append([a[1], a[0]])
+        cmap_new.sort(key=lambda x: x[0])
+        return cmap_new
+
+    def get_best_node(self):
+        '''
+        First node with the most connections; Does not yet sort
+        based on error, but that is probably not too useful
+        '''
+
+        bestNode = 0
+        for i in self.nodes:
```

Comment:
this can be a one-linear using numpy:
`bestNode = np.argmax([len(node) for node in self.nodes])`

Reply:
It's not the same here, a subtle difference in semantics. Doing it right would make it more cumbersome